pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -24,6 +24,40 @@ const (
 	PausedAnnotation = "baremetalhost.metal3.io/paused"
 )
 
+// RootDeviceHints holds the hints for specifying the storage location
+// for the root filesystem for the image.
+type RootDeviceHints struct {
+	// A device name like "/dev/vda"
+	DeviceName string `json:"devicename,omitempty"`
+
+	// A SCSI bus address like 0:0:0:0
+	HCTL string `json:"hctl,omitempty"`
+
+	// Device identifier
+	Model string `json:"model,omitempty"`
+
+	// Device vendor
+	Vendor string `json:"vendor,omitempty"`
+
+	// Disk serial number
+	Serial string `json:"serial,omitempty"`
+
+	// Size of the device in GiB
+	Size int `json:"size,omitempty"`
```

Comment:
Why GiB? Elsewhere (e.g. HardwareDetails) we always report storage sizes in bytes.
Also, we should include the units in the name.

Reply:
Changes to SizeGigaBytes. I would like to keep GS, as ironic takes GB as input, so I would avoid unit conversion. OK?